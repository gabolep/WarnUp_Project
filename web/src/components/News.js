import { Component } from 'react';
import New from './New';
import axios from 'axios';



const styles = {
    tabla: {
        backgroundColor: '#fff',
        position: 'static',
        marginTop: '40px',
        color: '#000000',
        cursor: 'pointer',
        

    },
    tr:{
        margin: 0,
        padding: 0,
        width:'100%'
    },
}
class News extends Component{
    state = {
        news: [],
    }
    componentDidMount(){
        axios.get('http://localhost:4000/new/').then(res => {
            this.setState({ news: res.data});
        });    
      }
   

    render(){
        return(
            <div style= {styles.tabla}>
                <table style= {styles.tr}>
                    <tbody>
                        { this.state.news.sort((act,pos) => {
                            const date1 = new Date(act.created_at);
                            const date2 = new Date(pos.created_at);
                            return date2.valueOf() - date1.valueOf()
                        }).map(n =>
                            <New key={n.story_id.toString()}
                                n = {n}
                            />
                        )}
                    </tbody>
                </table>
        </div>
        )}
}

export default News;
