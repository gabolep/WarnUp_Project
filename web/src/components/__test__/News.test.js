import react from "react";
import ReactDOM from "react-dom";
import News from '../News';
import { render, cleanup } from '@testing-library/react'
import renderer from "react-test-renderer";


afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<News></News>, div)
})
it("matches snapshot", () => {
    const tree = renderer.create(<News/>).toJSON();
    expect(tree).toMatchSnapshot();
})