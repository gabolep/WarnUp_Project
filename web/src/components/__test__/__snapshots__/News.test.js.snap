// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fff",
      "color": "#000000",
      "cursor": "pointer",
      "marginTop": "40px",
      "position": "static",
    }
  }
>
  <table
    style={
      Object {
        "margin": 0,
        "padding": 0,
        "width": "100%",
      }
    }
  >
    <tbody />
  </table>
</div>
`;
